### PR TITLE
[Action] Change ubuntu and yocto github action 

### PR DESCRIPTION
**Pull request by gichan-jang**
- Upload build result and badge of ubuntu and yocto.
 - Add condition to check the trigger event.

yocto daily build result: https://github.com/nnstreamer/nnstreamer/actions/runs/11789508454/job/32838369688

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: .github/workflows/yocto.yml [hunk 1]
```diff
@@ -47,7 +74,7 @@ jobs:
         echo "::group::Prepare poky and meta-neural-network"
         git clone git://git.yoctoproject.org/poky -b yocto-5.0.3 && cd poky
         git clone https://github.com/nnstreamer/meta-neural-network -b scarthgap
-        echo 'SRC_URI = "git://${{ github.workspace }}/;protocol=file;usehead=1;nobranch=1"' >> meta-neural-network/recipes-nnstreamer/nnstreamer/nnstreamer_%.bbappend
```
Comment (anyj0527) [inline .github/workflows/yocto.yml:50]: `nobranch=1` should be preserved.

Comment (gichan-jang) [reply .github/workflows/yocto.yml]: Thanks, Updated!

File: .github/workflows/yocto.yml [hunk 2]
```diff
@@ -76,13 +103,61 @@ jobs:
...
+        cp tmp/work/core2-64-poky-linux/nnstreamer/*/temp/log.do_compile ~/daily_build/logs/yocto_compile_log.txt
+        cp -r tmp/deploy ~/daily_build/deploy
+        echo "::endgroup::"
+
         sudo rm -rf tmp/
+
     - name: save yocto cache (main branch only)
       uses: actions/cache/save@v4
-      if: always() && github.ref == 'refs/heads/main'
+      if: steps.rebuild.outputs.rebuild == '1'
```
Comment (anyj0527) [inline .github/workflows/yocto.yml:117]: It's desired to save caches only for main branch. How about keep the old line?

Comment (gichan-jang) [reply .github/workflows/yocto.yml:117]: Right. Let's save cache only for main branch. Thanks.

File: .github/workflows/update_pbuilder_cache.yml [hunk 3]
```diff
@@ -59,23 +59,53 @@ jobs:
...
+          pdebuild --logfile ~/daily_build/pdebuild_log.txt --buildresult ~/daily_build/ubuntu --architecture ${{ matrix.arch }} -- --distribution ${{ matrix.distroname }}
+
+      - name: generate badge
+        run: |
+          pip install pybadges setuptools
+          if [ '${{ steps.pdebuid.outcome }}' == 'success' ]; then
+            python3 -m pybadges --left-text=test --right-text=success --right-color=green > ~/daily_build/pdebuild_result.svg
+          else
+            python3 -m pybadges --left-text=test --right-text=failure --right-color=red > ~/daily_build/pdebuild_result.svg
+          fi
```
Comment (songgot) [inline .github/workflows/update_pbuilder_cache.yml:75]: Good PR :+1: ,  Maybe we should change ```steps.pdebuid.outcome``` to ```steps.pdebuild.outcome```. Right?

Comment (gichan-jang) [reply .github/workflows/update_pbuilder_cache.yml]: Updated.

**Review by again4you — APPROVED** [review]

**Pull request merged**